tests/integration/test_data_store_mgr.py

```diff
@@ -313,12 +344,93 @@ async def test_update_data_structure(harness):
     assert TASK_STATUS_FAILED in set(collect_states(data, FAMILY_PROXIES))
     # state totals changed
     assert TASK_STATUS_FAILED in data[WORKFLOW].state_totals
-    # Shows pruning worked
-    # TODO: fixme
-    # https://github.com/cylc/cylc-flow/issues/4175#issuecomment-1025666413
-    # assert len({t.is_held for t in data[TASK_PROXIES].values()}) == 1
 
 
+async def test_prune_data_store(harness_prune):
+    """Test prune_data_store. This method will expand and reduce the data-store
+    to invoke pruning."""
```

Comment:
```suggestion
    to invoke pruning.

    https://github.com/cylc/cylc-ui/issues/1999
    """
```

Reply:
Have elaborated a little further, noting that the test covers the rapid addition/removal of a task (and it's families) with reference to the issue..